dcpy/lifecycle/builds/plan.py

```diff
@@ -23,6 +23,48 @@
 ]
 
 
+def resolve_version(recipe: Recipe) -> str:
+    match recipe.version_strategy:
+        case None:
+            raise Exception("No version or version_strategy provided")
+        case versions.SimpleVersionStrategy.first_of_month:
+            return versions.generate_first_of_month().label
+        case versions.SimpleVersionStrategy.bump_latest_release:
+            previous_version = publishing.get_latest_version(recipe.product)
+            assert previous_version is not None
+            return versions.bump(
+                previous_version=previous_version,
+                bump_type=recipe.version_type,
+            ).label
+        case versions.BumpLatestRelease() as bump:
+            previous_version = publishing.get_latest_version(recipe.product)
+            assert previous_version is not None
+            return versions.bump(
+                previous_version=previous_version,
+                bump_type=recipe.version_type,
+                bump_by=bump.bump_latest_release,
+            ).label
+        case versions.PinToSourceDataset():
```

Comment:
codecov thinks this line is only partially covered - I think by their syntax its really the `match` statement. As in, it isn't covered when none of these match. Unfortunately for codecov, this matching is exhaustive (which mypy would catch if not). Don't really want to add an unreachable `case _:` just to make codecov happier